### ISSUE TITLE
Fix S2583: Rule should doesn't detect try/finally properly

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#617]
### Description

Currently the rule suggest to correct the content of the `finally` block which is always `false`. This is a FP as `fileStream` is actually `null` in most of the cases.

### Repro steps

```
StringBuilder contentRaw = new StringBuilder();
Stream fileStream = null;

try
{
    fileStream = File.Open("MyFile.txt", FileMode.Open, FileAccess.Read);
    using (StreamReader reader = new StreamReader(fileStream, Encoding.UTF8))
    {
        fileStream = null;
        while (!reader.EndOfStream)
        {
            contentRaw.AppendLine(reader.ReadLine());
        }
    }
}
finally
{
    if (fileStream != null) // Raise S2583
    {
        fileStream.Close();
    }
}
```

### Expected behavior

Rule should not raise an issue.

### Actual behavior

Rule raises an issue.

**Comment by Evangelink** [comment]
This is linked to a well-known limitation of our CFG implementation around try/catch/finally that we don't plan to fix soon.
